cpp/libc.cc

```diff
@@ -163,4 +165,33 @@ Tuple2<int, int>* regex_first_group_match(Str* pattern, Str* str, int pos) {
   return tup;
 }
 
+int wcswidth(Str* str) {
+  int len = mbstowcs(NULL, str->data(), 0);
+  if (len == -1) {
+    throw Alloc<UnicodeError>(StrFromC("mbstowcs error: Invalid UTF-8 string"));
+  }
+
+  auto* unicode = static_cast<wchar_t*>(malloc(len + 1));
+  if (unicode == nullptr) {
+    // NOTE: glibc malloc() sets errno to ENOMEM on failure, but customer
```

Comment:
I think we should either just `assert(unicode != nullptr)` or throw a custom exception and for allocation failure and handle it at the top level, with a custom message

Right now we are using malloc() in a few places, and those aren't checked

I think this is conceptually distinct and much rarer than IOError/OSError, so I'm OK with deferring it for awhile